src/components/displayAllComponent.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from "react";
+import ChooseFolder from "./chooseFolder/chooseFolder";
+import { Redirect } from "react-router-dom";
+import SvgSettingOptions from "./svgSetting/svgSettingOptions";
+import "../components/checkBoxSelection/checkBoxSelection.scss";
+import { changeObj, optimizeSvg } from "../../src/components/functions";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { UPDATE_VARIABLE} from "../../src/store/actionTypes";
+
+class DisplayAllComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      redirect: false
+        };
+  }
+  getOptimizeSvg = async () => {
+    let originalAndOptimised = {};
+    let tempOptimisationArr = [];
+    for (var option of this.props.svgOptions) {
+      changeObj(option, this.props.svgObject.plugins);
+    }
+    for (var setting of this.props.svgSettingList) {
+      if (this.props.svgOptions.length > 0) {
+        originalAndOptimised = {
+          originalSvg: setting.dataurl,
+          name: setting.name,
+          optimisedSvg: await optimizeSvg(
+            setting.dataurl,
+            this.props.svgObject.plugins
+          )
+        };
+      } else {
+        originalAndOptimised = {
+          originalSvg: setting.dataurl,
+          name: setting.name,
+          optimisedSvg: await optimizeSvg(
+            setting.dataurl,
+            this.props.svgObject.plugins
+          )
+        };
```

Comment:
both of these look exactly the same, maybe I'm missing something?

Reply:
the other second one suppose to take  a null : `await optimizeSvg(
            setting.dataurl,
            null
          )`

Reply:
its still alot of duplicated code tho, view my comment below for a suggestion

Reply:
I have used the similar approach to the of your suggestion below.